docs/kafka.md

```diff
@@ -0,0 +1,92 @@
+
+
+Example Configuration:
+
+```yaml
+metrics:
+  receivers:
+    kafka_metrics:
+      type: kafka
+      endpoint: localhost:9999
+      collection_interval: 30s
+  service:
+    pipelines:
+      kafka_pipeline:
+        receivers:
+          - kafka_metrics
+```
+
+## Metrics
+In addition to Kafka specific metrics, by default Kafka will also report [JVM metrics](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/docs/jvm.md#metrics)
+
+| Metric                                                    | Data Type | Unit        | Labels | Description |
+| ---                                                       | ---       | ---         | ---    | ---         | 
+| workload.googleapis.com/kafka.message.count               | cumulative | messages | operation | The number of messages received by the broker |
+| workload.googleapis.com/kafka.request.count               | cumulative | requests          | type | The number of requests received by the broker |
+| workload.googleapis.com/kafka.request.failed              | cumulative | requests          | type | The number of requests to the broker resulting in a failure |
+| workload.googleapis.com/kafka.request.time.total          | cumulative | ms          | type | The total time the broker has taken to service requests |
+| workload.googleapis.com/kafka.network.io                  | cumulative | by          | state | The bytes received or sent by the broker |
+| workload.googleapis.com/kafka.purgatory.size              | gauge | requests          | type | The number of requests waiting in purgatory |
+| workload.googleapis.com/kafka.partition.count             | gauge | partitions          |  | The number of partitions on the broker |
+| workload.googleapis.com/kafka.partition.offline           | gauge | partitions          |  | The number of partitions offline |
+| workload.googleapis.com/kafka.partition.under_replicated  | gauge | partitions          |  | The number of under replicated partitions |
+| workload.googleapis.com/kafka.isr.operation.count         | cumulative | 1          | operation | The number of in-sync replica shrink and expand operations |
+
+
+# `kafka` Logging Receiver
+
+## Configuration
+
+To configure a receiver for your kafka logs, specify the following fields:
+
+| Field                 | Default                       | Description |
+| ---                   | ---                           | ---         |
+| `type`                | required                      | Must be `kafka`. |
+| `include_paths`       | `[/var/log/kafka/*.log]` | A list of filesystem paths to read by tailing each file. A wild card (`*`) can be used in the paths; for example, `/var/log/kafka*/*.log`.
+| `exclude_paths`       | `[]`                          | A list of filesystem path patterns to exclude from the set matched by `include_paths`.
+
+Example Configuration:
+
+```yaml
+logging:
+  receivers:
+    kafka_default:
```

Comment:
It's preferred for the receiver and pipeline IDs to stay the same as the receiver type in the sample configuration. E.g.  "s/{APP}_metrics/{APP}", "s/{APP}_pipeline/{APP}", "s/{APP}_default/{APP}"